src/client-registry/client-registry-data/client-registry-data.component.tsx

```diff
@@ -0,0 +1,238 @@
+import {
+  ExtensionSlot,
+  formatDate,
+  parseDate,
+  showNotification,
+  showSnackbar,
+  usePagination,
+} from "@openmrs/esm-framework";
+import {
+  DataTable,
+  TableContainer,
+  Pagination,
+  Tile,
+  TableBody,
+  Table,
+  TableHead,
+  TableRow,
+  TableHeader,
+  TableCell,
+  DataTableSkeleton,
+  Button,
+  InlineLoading,
+} from "@carbon/react";
+import styles from "./client-registry-data.scss";
+import { OverflowMenuVertical } from "@carbon/react/icons";
+
+import OrderCustomOverflowMenuComponent from "../../ui-components/overflow-menu.component";
+
+const ClientRegistryData: React.FC = () => {
+  const { t } = useTranslation();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { patients, isLoading: loading, mutate } = usePatients("sarah", false);
+
+  const pageSizes = [10, 20, 30, 40, 50];
+  const [currentPageSize, setPageSize] = useState(10);
+  const {
+    goTo,
+    results: paginatedPatientEntries,
+    currentPage,
+  } = usePagination(patients, currentPageSize);
+
+  const tableHeaders = useMemo(
+    () => [
+      { id: 0, header: t("name", "Name"), key: "name" },
+      { id: 1, header: t("identifiers", "Identifiers"), key: "identifiers" },
+      { id: 2, header: t("gender", "Gender"), key: "gender" },
+      { id: 3, header: t("age", "Age"), key: "age" },
+      { id: 4, header: t("birthdate", "Birthdate"), key: "birthdate" },
+      {
+        id: 5,
+        header: t("maritalStatus", "Marital Status"),
+        key: "maritalStatus",
+      },
+      { id: 6, header: t("occupation", "Occupation"), key: "occupation" },
+      { id: 7, header: t("nationality", "Nationality"), key: "nationality" },
+      { id: 8, header: t("dead", "Dead"), key: "dead" },
+      { id: 9, actions: t("actions", "Actions"), key: "actions" },
+    ],
+    [t]
+  );
+
+  const startClientRegistry = async (e) => {
+    e.preventDefault();
+    setIsLoading(true);
+    startClientRegistryTask().then(
+      () => {
+        mutate();
+        setIsLoading(false);
+        showSnackbar({
+          isLowContrast: true,
+          title: t("runTask", "Start Client Registry Task"),
+          kind: "success",
+          subtitle: t(
+            "successfullyStarted",
+            `You have successfully started Client Registry`
+          ),
+        });
+      },
+      (error) => {
+        const errorMessages = extractErrorMessagesFromResponse(error);
+
+        mutate();
+        setIsLoading(false);
+        showNotification({
+          title: t(
+            `errorStartingTask', 'Error Starting client registry task"),`
+          ),
+          kind: "error",
+          critical: true,
+          description: errorMessages.join(", "),
+        });
+      }
+    );
+  };
+
+  const tableRows = useMemo(() => {
+    return patients?.map((patient, index) => ({
+      ...patient,
+      id: patient?.uuid,
+      name: patient?.person?.display,
+      identifiers: patient.identifiers
```

Comment:
It would be better to have a mapFunction in a resource file to handle all this row logic so that you lift the weight in this file

Reply:
we shall do a clean sweep later on this

Reply:
and this is also not doing alot